cdap-security/src/main/java/co/cask/cdap/security/impersonation/SecurityUtil.java

```diff
@@ -278,4 +279,28 @@ public static void verifyOwnerPrincipal(NamespacedEntityId existingEntity,
                                                     Constants.Security.PRINCIPAL));
     }
   }
+
+  /**
+   * Helper function to get the effective owner of an entity. It will check the owner store to get the namespace
+   * owner if the provided owner principal is null.
+   *
+   * Note that this function should only be used when we
+   * do not put the owner of the entity, if there is, to the owner store.
+   *
+   * @param ownerAdmin owner admin to query the owner
+   * @param namespaceId the namespace the entity is in
+   * @param ownerPrincipal the owner principal of the entity, null if not provided
+   * @return the effective owner of the entity, null if no owner is provided for both the enity and the namespace.
+   */
+  @Nullable
+  public static KerberosPrincipalId getEffectiveOwner(OwnerAdmin ownerAdmin, NamespaceId namespaceId,
+                                                      String ownerPrincipal) throws IOException {
+    String effectiveOwner = ownerPrincipal == null ? null : ownerPrincipal;
```

Comment:
This can be simplified as - 

```
if (ownerPrincipal != null) {
  // if entity owner is present, return it
  return;
}

if (!namespaceId.equals(NamespaceId.SYSTEM)) {
  // if entity owner is not present, get the namespace impersonation principal
  String namespacePrincipal = ownerAdmin.getImpersonationPrincipal(namespaceId);
  return new KerberosPrincipalId(namespacePrincipal);
}

// No owner found
return null;
```